tasks/deployment/config.ts

```diff
@@ -0,0 +1,177 @@
+import { task } from 'hardhat/config'
+import '@nomiclabs/hardhat-ethers'
+import { cliOpts } from '../../cli/defaults'
+import { readConfig, writeConfig } from '../../cli/config'
+import YAML from 'yaml'
+
+import { Scalar, YAMLMap } from 'yaml/types'
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+import fs from 'fs'
+import inquirer from 'inquirer'
+
+interface Contract {
+  name: string
+  initParams: ContractInitParam[]
+}
+
+interface ContractInitParam {
+  name: string // as declared in config.yml
+  type: 'number' | 'BigNumber' // as returned by the contract
+  getter?: string // name of function to get the value from the contract. Defaults to 'name'
+  format?: 'number' // some parameters are stored in different formats than what the contract reports.
+}
+
+const epochManager: Contract = {
+  name: 'EpochManager',
+  initParams: [
+    { name: 'lengthInBlocks', type: 'BigNumber', getter: 'epochLength', format: 'number' },
+  ],
+}
+
+// GraphToken
+// TODO > 'initialSupply' not on chain
+// const graphToken: Contract = {
+//   name: 'GraphToken',
+//   initParams: [],
+// }
+
+const curation: Contract = {
```

Comment:
You could import the ABIs JSON from the `build/abi` folder to avoid copying them over.

Reply:
@abarmat I thought about using ABIs but how would you filter out all the stuff that's not related to init params? You'd still need to hardcode a list right?

Reply:
@tmigone I get it, you are using the list to know what to scan, all good!